tests/watchFiles.spec.ts

```diff
@@ -0,0 +1,38 @@
+import fs from 'fs/promises';
```

Comment:
You may want to try doing e2e tests in the future, creating real files, moving them and then checking they are in the right place. That avoids just testing the implementation details which you are doing now.